processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -51,25 +60,44 @@ public SelectorDimFilter(
   {
     byte[] dimensionBytes = StringUtils.toUtf8(dimension);
     byte[] valueBytes = (value == null) ? new byte[]{} : StringUtils.toUtf8(value);
+    byte[] extractionFnBytes = extractionFn == null ? new byte[0] : extractionFn.getCacheKey();
 
-    return ByteBuffer.allocate(2 + dimensionBytes.length + valueBytes.length)
+    return ByteBuffer.allocate(3 + dimensionBytes.length + valueBytes.length + extractionFnBytes.length)
                      .put(DimFilterCacheHelper.SELECTOR_CACHE_ID)
                      .put(dimensionBytes)
                      .put(DimFilterCacheHelper.STRING_SEPARATOR)
                      .put(valueBytes)
+                     .put(DimFilterCacheHelper.STRING_SEPARATOR)
+                     .put(extractionFnBytes)
                      .array();
   }
 
   @Override
   public DimFilter optimize()
   {
+    if (this.getExtractionFn() instanceof LookupExtractionFn
+        && ((LookupExtractionFn) this.getExtractionFn()).isOptimize()) {
+      LookupExtractor lookup = ((LookupExtractionFn) this.getExtractionFn()).getLookup();
+      final List<String> keys = lookup.unapply(this.getValue());
+      final String dimensionName = this.getDimension();
+      if (!keys.isEmpty()) {
+        return new OrDimFilter(Lists.transform(keys, new Function<String, DimFilter>()
```

Comment:
This could be an InDimFilter (it looks like you copied this from ExtractionDimFilter, which was recently updated to use that)